### ISSUE TITLE
More idempotent search

### ISSUE THREAD

**Issue by Timalex** [solus-project/solus-sc#13]
If selecting "Search" again, always display the search field immediately (requires using back button now).

**Comment by ikeydoherty** [comment]
Nothing like issue #1, that described the old SC. The search field doesn't get hidden here I can't see what the issue is

**Comment by Timalex** [comment]
1. Open SC
2. Select **Search**. _The search field is visible._
3. Find a package and "open" it. _The search field is "hidden"._
4. Do something else e.g. **Updates** > **Check for updates**
5. Select **Search**. _The search field is "hidden"._

Compare step 2 and 5. 
So __ability to search is not visible after selecting Search__, in the second instance. 
When selecting Search in step 5. It should either be reset and rendered exactly as in step 2 or just as if the back button had been pressed to retain some information about the previous search.

**Comment by ikeydoherty** [comment]
The search field is within its own page, it's not a global search field. This is _by design_.

**Comment by Timalex** [comment]
Let me rephrase that then.
When selecting **Search** from the sidebar in SC, at step 5, it could simply "reset" to display an existing page that contains the search field.

**Comment by baimafeima** [comment]
Timalex' suggestion would save one click. 

> The search field is within its own page, it's not a global search field. This is by design.

What is the advantage of this? Maybe there is something I've overlooked?
